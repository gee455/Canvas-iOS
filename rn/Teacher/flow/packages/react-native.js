//
// Copyright (C) 2016-present Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, version 3 of the License.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.
//

// @flow

declare var console: typeof console & { disableYellowBox: boolean }

export type PanResponderGestureState = {
  stateID: number,
  moveX: number,
  moveY: number,
  x0: number,
  y0: number,
  dx: number,
  dy: number,
  vx: number,
  vy: number,
}

export type KeyboardEventData = {
  endCoordinates: {
    width: number,
    height: number,
    screenX: number,
    screenY: number,
  },
}
