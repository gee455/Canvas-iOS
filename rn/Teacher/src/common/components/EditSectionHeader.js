//
// Copyright (C) 2016-present Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, version 3 of the License.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.
//

/**
 * @flow
 */

import React, { PureComponent } from 'react'
import { Heading1 } from '../text'
import color from '../colors'
import {
  View,
  StyleSheet,
} from 'react-native'

export default class EditSectionHeader extends PureComponent<*> {
  render () {
    return (
      <View style={[style.container, this.props.style]}>
        <Heading1 style={style.header}>{this.props.title}</Heading1>
        {this.props.children}
      </View>
    )
  }
}

const style = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'flex-start',
    alignItems: 'center',
    height: 'auto',
    backgroundColor: '#F5F5F5',
  },
  header: {
    color: color.darkText,
    marginLeft: global.style.defaultPadding,
    marginTop: global.style.defaultPadding,
    marginBottom: global.style.defaultPadding / 2,
  },
})
