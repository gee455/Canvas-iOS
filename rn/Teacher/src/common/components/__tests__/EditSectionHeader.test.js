//
// Copyright (C) 2016-present Instructure, Inc.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//

/**
 * @flow
 */

import 'react-native'
import React from 'react'
import EditSectionHeader from '../EditSectionHeader'
import renderer from 'react-test-renderer'

const defaultProps = {
  title: 'foo',
}

test('render', () => {
  let tree = renderer.create(
    <EditSectionHeader {...defaultProps} />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('render no title', () => {
  let props = Object.assign({}, defaultProps)
  delete props.title
  let tree = renderer.create(
    <EditSectionHeader {...props} />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
