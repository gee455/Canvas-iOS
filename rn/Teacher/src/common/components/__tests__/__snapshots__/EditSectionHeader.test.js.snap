// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#F5F5F5",
        "flex": 1,
        "flexDirection": "row",
        "height": "auto",
        "justifyContent": "flex-start",
      },
      undefined,
    ]
  }
>
  <Text
    accessibilityTraits="header"
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontFamily": ".SFUIDisplay",
        },
        Object {
          "color": "#2D3B45",
          "fontFamily": ".SFUIDisplay-semibold",
          "fontSize": 20,
        },
        Object {
          "color": "#2D3B45",
          "marginBottom": 8,
          "marginLeft": 16,
          "marginTop": 16,
        },
      ]
    }
  >
    foo
  </Text>
</View>
`;

exports[`render no title 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#F5F5F5",
        "flex": 1,
        "flexDirection": "row",
        "height": "auto",
        "justifyContent": "flex-start",
      },
      undefined,
    ]
  }
>
  <Text
    accessibilityTraits="header"
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontFamily": ".SFUIDisplay",
        },
        Object {
          "color": "#2D3B45",
          "fontFamily": ".SFUIDisplay-semibold",
          "fontSize": 20,
        },
        Object {
          "color": "#2D3B45",
          "marginBottom": 8,
          "marginLeft": 16,
          "marginTop": 16,
        },
      ]
    }
  />
</View>
`;
